### PR TITLE
Webpack bundling

### PR DESCRIPTION
**Pull request by bre1470**
Testing:

`npx gulp scripts --webpack`
`npx gulp test --webpack`

Please note that declarations are not generated (yet) by webpack.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206009512469955

### CODE REVIEW

**Comment by TorsteinHonsi** [comment]
The CI doesn't run, and it doesn't run locally on a fresh system. It stops of "Generating code..." because the Webpack CLI is not installed. This is how it looks when running the task directly, but the yes/no prompt is not revealed when running `.exec` trough the gulp task.

Skjermbilde 2023-11-29 kl  12 22 01

**Review by goransle — APPROVED** [review]
Besides those hiccups I don't see anything blocking a merge 👍

**Comment by KacperMadej** [comment]
`npx gulp test --webpack` fails for me with error:
```console
Chrome Headless 121.0.6167.185 (Windows 10) ERROR
  Uncaught TypeError: Cannot read properties of undefined (reading 'prototype')
  at code/highcharts-more.src.js:19346:28

  TypeError: Cannot read properties of undefined (reading 'prototype')
      at code/highcharts-more.src.js:19346:28
      at code/highcharts-more.src.js:45013:12
      at webpackUniversalModuleDefinition (code/highcharts-more.src.js:9:24)
      at code/highcharts-more.src.js:10:3
```

that points to the first `prototype` in this line of code:
```
const { area: { prototype: { pointClass: AreaPoint, pointClass: { prototype: areaProto } } } } = Series_SeriesRegistry.seriesTypes;
```

**Comment by TorsteinHonsi** [comment]
Before merging to master, shouldn't we
* Include Dashboard builds?
* Remove the flags (`npx gulp test --webpack` etc), as webpack becomes standard?

**Comment by bre1470** [comment]
This is not production ready.

**Comment by bre1470** [comment]
The point to merge this into master:
- Limit the already huge scope of review
- Avoid further merge conflicts
- Have more setups to test against
- Optimize webpack
- Prepare integration of webpack

**Comment by bre1470** [comment]
@KacperMadej I fixed the Windows-compatibility of the Webpack configuration. Please review ones more.

File: tools/webpacks/highcharts.webpack.mjs
Comment (TorsteinHonsi) [inline tools/webpacks/highcharts.webpack.mjs:1]: Is this file used anymore? Or just a remnant of the POC phase?

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks for the solid piece of work 💪 !

File: tools/webpacks/highcharts.webpack.mjs
Comment (bre1470) [reply tools/webpacks/highcharts.webpack.mjs:1]: This is the configuration for webpack.

Comment (TorsteinHonsi) [reply tools/webpacks/highcharts.webpack.mjs:1]: Okay! Should it be `/webpacks` though? Plural?

Comment (bre1470) [reply tools/webpacks/highcharts.webpack.mjs:1]: There is an old MVP-variant for Dashboards as well. Basically base products need their own webpack configuration, as internal and external reference together with the modules organisation differ.

**Comment by KacperMadej** [comment]
> @KacperMadej I fixed the Windows-compatibility of the Webpack configuration. Please review ones more.

It works for me ✅

**Comment by TorsteinHonsi** [comment]
Oh. There's a conflict...

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**